src/components/common/_material-overrides-buttons.scss

```diff
@@ -0,0 +1,293 @@
+
+    // .mdc-button__label {
+    //   color: $foreColor;
+    //   z-index: 2; // white text does not get darkened by ::after overlay
+    // }
+
+    &:before,
+    &:after {
+      background-color: $bgColorHoverFocus;
+    }
+
+    &:hover {
+      &:before,
+      &:after {
+        opacity: 0.12;
+      }
+    }
+
+    &:focus:not(:hover):not(:active) {
+      background-color: transparent;
+    }
+
+    &:active {
+      &:before,
+      &:after {
+        opacity: 0.24;
+      }
+    }
+  }
+
+  &:disabled {
+    color: $disabledForeColor !important; // needed to override mdc built in
+  }
+
+  // Icons inside flat buttons
+  md-icon,
+  .fb-icon-animation-container,
+  .fire-icon-animation-container {
+    background-color: transparent;
+    color: currentColor;
+    fill: currentColor;
+  }
+}
+
+// Flat primary
+@include __for-text-buttons-of-type(flat, primary) {
+  @include __flat-text-button-styling(
+    $foreColor: var(--mdc-theme-primary),
+    $bgColorHoverFocus: var(--mdc-theme-primary)
+  );
+}
+
+// Flat secondary (non-primary)
+@include __for-text-buttons-of-type(flat, secondary) {
+  @include __flat-text-button-styling(
+    $foreColor: $COLOR_BLACK_SECONDARY,
+    $bgColorHoverFocus: $COLOR_PURE_BLACK
+  );
+}
+
+.mdc-theme--on-primary {
+  // Dark raised primary buttons
+  @include __for-text-buttons-of-type(raised, primary) {
+    @include __raised-text-button-styling(
+      $foreColor: var(--mdc-theme-primary),
+      $bgColorDefault: $COLOR_WHITE_PRIMARY,
+      $bgColorHoverFocus: $COLOR_PURE_BLACK,
+      $disabledBgColor: rgba($COLOR_WHITE_PRIMARY, 0.12),
+      $disabledForeColor: $COLOR_WHITE_DISABLED,
+      $insetFocusOutlineColor: rgba(var(--mdc-theme-primary-rgb), 0.5)
+    );
+  }
+
+  // Dark raised secondary + all hairline buttons
+  %all-dark-hairline-buttons {
+    @include __raised-text-button-styling(
+      $foreColor: $COLOR_WHITE_PRIMARY,
+      $bgColorDefault: transparent,
+      $bgColorHoverFocus: $COLOR_WHITE_PRIMARY,
+      $hairlineShadowDefault: 0 0 0 1px rgba($COLOR_WHITE_PRIMARY, 0.5) inset,
+      $hairlineShadowHoverFocus: 0 0 0 1px rgba($COLOR_WHITE_PRIMARY, 0.8) inset,
+      $hairlineShadowActive: 0 0 0 1px $COLOR_WHITE_PRIMARY inset,
+      $hairlineShadowDisabled: 0 0 0 1px rgba($COLOR_WHITE_PRIMARY, 0.2) inset,
+      $disabledBgColor: transparent,
+      $disabledForeColor: $COLOR_WHITE_DISABLED
+    );
+  }
+
+  @include __for-text-buttons-of-type(raised, secondary) {
+    @extend %all-dark-hairline-buttons;
+  }
+
+  @include __for-text-buttons-of-type(stroked, primary) {
+    @extend %all-dark-hairline-buttons;
+  }
+
+  @include __for-text-buttons-of-type(stroked, secondary) {
+    @extend %all-dark-hairline-buttons;
+  }
+
+  // Dark flat primary, secondary, and warn (all the same)
+  %all-dark-flat-buttons {
+    @include __flat-text-button-styling(
+      // sorry about important, overrides mdc important
+        $foreColor: $COLOR_WHITE_PRIMARY !important,
```

Comment:
```suggestion
      $foreColor: $COLOR_WHITE_PRIMARY !important,
```

Reply:
It's auto formatted, so it reverts. will leave :)